hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/Enhancer.java

```diff
@@ -107,11 +105,8 @@ private ClassPool buildClassPool(EnhancementContext enhancementContext) {
 	}
 
 	private CtClass loadCtClassFromClass(ClassPool cp, Class<?> aClass) throws IOException {
-		return cp.makeClass( aClass.getClassLoader().getResourceAsStream( getFilenameForClass( aClass ) ) );
-	}
-
-	private String getFilenameForClass(Class<?> aClass) {
-		return aClass.getName().replace( '.', File.separatorChar ) + JavaFileObject.Kind.CLASS.extension;
+		String resourceName = aClass.getName().replace( '.', '/' ) + ".class";
+		return cp.makeClass( aClass.getClassLoader().getResourceAsStream( resourceName ) );
```

Comment:
I wonder what closes the stream returned by getResourceAsStream()?  Should we close that?